crypto/digestible/derive/src/lib.rs

```diff
@@ -82,6 +78,126 @@ fn try_digestible(input: TokenStream) -> Result<TokenStream, &'static str> {
     Ok(expanded.into())
 }
 
+fn try_digestible_enum(
+    ident: &Ident,
+    generics: &Generics,
+    variant_data: &DataEnum,
+) -> Result<TokenStream, &'static str> {
+    let call = variant_data
+        .variants
+        .iter()
+        .map(|variant| {
+            let variant_ident = &variant.ident;
+
```

Comment:
I think I would want to add something like, `variants.enumerate()` so that we know if this is the `i`'th possibility, and then encode that number as a fixed width integer, e.g. a u32, and stick that in the hash before the other stuff. Something that is ensuring that there is a prefix-free representation of which variant it is before we start hitting the data associated to that value. Because the enum variant names are user generated and may not be fixed width or prefix free -- we'd like to ensure that there's no way that a hash from Option1 and Option2 can collide